core/src/main/scala/org/apache/spark/api/r/SerDe.scala

```diff
@@ -272,18 +282,22 @@ private[spark] object SerDe {
     }
   }
 
-  private def writeKeyValue(dos: DataOutputStream, key: Object, value: Object): Unit = {
+  private def writeKeyValue(
+      dos: DataOutputStream,
+      key: Object,
+      value: Object,
+      jvmObjectTracker: JVMObjectTracker): Unit = {
     if (key == null) {
       throw new IllegalArgumentException("Key in map can't be null.")
     } else if (!key.isInstanceOf[String]) {
       throw new IllegalArgumentException(s"Invalid map key type: ${key.getClass.getName}")
     }
 
     writeString(dos, key.asInstanceOf[String])
-    writeObject(dos, value)
+    writeObject(dos, value, jvmObjectTracker)
   }
 
-  def writeObject(dos: DataOutputStream, obj: Object): Unit = {
+  def writeObject(dos: DataOutputStream, obj: Object, jvmObjectTracker: JVMObjectTracker): Unit = {
```

Comment:
I dont know if we have a policy, but I for one like the explicit parameter in this case. It at least makes it clear where we need to reason about the JVM object tracker being used.  Also if somebody adds a new `SerDe.writeObject` call in the future it ensures they think about it.